osvr/RenderKit/RenderManagerD3DBase.cpp

```diff
@@ -180,6 +182,57 @@ namespace renderkit {
         delete m_library.D3D11;
     }
 
+    bool RenderManagerD3D11Base::PresentRenderBuffersInternal(
+        const std::vector<RenderBuffer>& buffers,
+        const std::vector<RenderInfo>& renderInfoUsed,
+        const RenderParams& renderParams,
+        const std::vector<OSVR_ViewportDescription>& normalizedCroppingViewports,
+        bool flipInY) {
+
+        HRESULT hr;
+        {
+            std::set<ID3D11Texture2D*> lockedTextures;
+            for (size_t i = 0; i < buffers.size(); i++) {
+                auto colorBuffer = buffers[i].D3D11->colorBuffer;
+                if (lockedTextures.find(colorBuffer) == lockedTextures.end()) {
+                    lockedTextures.insert(colorBuffer);
+                    D3D11_TEXTURE2D_DESC desc = { 0 };
+                    colorBuffer->GetDesc(&desc);
+                    if ((desc.MiscFlags & D3D11_RESOURCE_MISC_SHARED_KEYEDMUTEX) == D3D11_RESOURCE_MISC_SHARED_KEYEDMUTEX) {
+                        IDXGIKeyedMutex* mutex = nullptr;
+                        hr = colorBuffer->QueryInterface(__uuidof(IDXGIKeyedMutex), (LPVOID*)&mutex);
+                        if (!FAILED(hr) && mutex != nullptr) {
+                            hr = mutex->AcquireSync(0, 500); // ignore failure
```

Comment:
Will fix, but I'm curious: what happens differently in a 500ms wait vs a 0-time wait? Surely it doesn't wait for the entire 500ms if it can't unlock immediately? The docs seem to indicate that any mutex waiting for unlock will get woken up, but maybe that only applies to an INFINITE wait?